lib/pdk/cli/validate.rb

```diff
@@ -70,15 +71,43 @@ module PDK::CLI
 
       options = targets.empty? ? {} : { targets: targets }
       options[:auto_correct] = true if opts.key?(:'auto-correct')
-
-      exit_code = 0
+      options[:parallel] = true if opts.key?(:parallel)
 
       # Ensure that the bundle is installed and tools are available before running any validations.
       PDK::Util::Bundler.ensure_bundle!
 
-      validators.each do |validator|
-        exit_code = validator.invoke(report, options)
-        break if exit_code != 0
+      PDK::CLI::Util.ensure_in_module!
+
+      exit_code = 0
+      if opts[:parallel]
+        threaded_spinner = PDK::CLI::Spinner.new_validation_spinner(validators.count, options)
+
+        threads = []
+        exit_codes = []
+        validators.each do |validator|
+          Dir.chdir(PDK::Util.module_root) do
+            threads << Thread.new do
+              require 'pdk/i18n'
+              exit_codes << validator.invoke(report, options)
+            end
+          end
+        end
+
+        threads.each(&:join)
+        exit_codes.sort!
+
+        if exit_codes.last != 0
```

Comment:
So that also needs fixing then: http://ruby-doc.org/core-2.4.0/Enumerable.html#method-i-max